File: backend/Models/Enums/OrderStatus.cs
namespace backend.Models.Enums;

public enum OrderStatus
{
    Pending = 0,
    Paid = 1,
    Shipped = 2,
    Cancelled = 3,

    /// <summary>ลูกค้าขอยกเลิก — รอแอดมินหรือเจ้าของสินค้าในออเดอร์พิจารณา</summary>
    CancellationPending = 4,
}

File: frontend/src/app/core/utils/order-status.ts
/** ค่าตรงกับ backend OrderStatus */
export function orderStatusLabelTh(status: number): string {
  switch (status) {
    case 0:
      return 'รอชำระเงิน';
    case 1:
      return 'ชำระแล้ว';
    case 2:
      return 'จัดส่งแล้ว';
    case 3:
      return 'ยกเลิก';
    case 4:
      return 'รอยกเลิก (รอพิจารณา)';
    default:
      return String(status);
  }
}
